File: fsharp/anagram/Anagram.fs
module Anagram

open System

let getCharSet word = 
    word |> Seq.countBy (fun c -> Char.ToLower c) |> Set.ofSeq

let getLower word =
    word |> Seq.map (Char.ToLower) |> Seq.toArray |> String

let isAnagram targetSet source =
    getCharSet source = targetSet

let findAnagrams sources target = 
    let targetSet = getCharSet target
    let isOtherWord source = (target |> getLower) <> (source |> getLower)
    sources |> List.filter isOtherWord |> List.filter (isAnagram targetSet) 

File: csharp/anagram/Anagram.cs
using System;
using System.Linq;
public class Anagram
{
    public readonly string _target;
    public readonly string _sortedTarget;
    public Anagram(string baseWord) { _target = baseWord.ToLower(); _sortedTarget = SortedLower(_target); }
    public string[] FindAnagrams(string[] potentialMatches) => 
        potentialMatches.Where(w => !w.ToLower().Equals(_target) && SortedLower(w).Equals(_sortedTarget) ).ToArray();
    public string SortedLower(string word) =>  new string(word.ToLower().OrderBy(c => c).ToArray());
}

